src/join.rs

```diff
@@ -77,7 +78,13 @@ pub trait Join {
     {
         JoinIter::new(self)
     }
-
+    /// Create a joined parallel iterator over the contents.
+    #[cfg(feature="parallel")]
+    fn par_join(self) -> JoinParIter<Self>
```

Comment:
@WaDelma Yeah, that's what I thought. Just a quick idea, I thought it might be easier then.